cmd/podman/pull.go

```diff
@@ -107,18 +118,58 @@ func pullCmd(c *cliconfig.PullValues) error {
 		if err != nil {
 			return errors.Wrapf(err, "error pulling image from %q", image)
 		}
-		imgID = newImage[0].ID()
+		fmt.Println(newImage[0].ID())
 	} else {
-		authfile := getAuthFile(c.Authfile)
-		newImage, err := runtime.New(getContext(), image, c.SignaturePolicy, authfile, writer, &dockerRegistryOptions, image2.SigningOptions{}, true, nil)
+		authfile := getAuthFile(c.String("authfile"))
```

Comment:
Sorry for the late-breaking request, but would it make more sense to move most of this block into `ImageRuntime.New()` (add an all-tags bool, and handle making possible names there?). Would require us to return an array of images instead, which would always be a single image if the all-tags bool was false...

Maybe it would be better to have an `ImageRuntime.PullAllTags()` or similar?

@baude WDYT?

Reply:
@baude, thoughts? 

Personally I'd vote to get this in now then I'd be happy to rework separately.  If the consensus is to get it all done here in this PR, I can start working on it.

Personally I don't see many people using this option.